### PR TITLE
HBP-95: Enable Upload of Encrypted data files in TVB Web GUI - COPY

### PR DESCRIPTION
**Pull request by VinczeRobert**
This is a copy of this closed PR https://github.com/the-virtual-brain/tvb-root/pull/205 and it is (in my opinion) ready to be merged.

### CODE REVIEW

File: framework_tvb/tvb/core/adapters/abcuploader.py [hunk 1]
```diff
@@ -69,6 +84,9 @@ class ABCUploader(ABCSynchronous, metaclass=ABCMeta):
     """
     LOGGER = get_logger(__name__)
 
+    def __init__(self):
+        super(ABCUploader, self).__init__()
```
Comment (liadomide) [inline framework_tvb/tvb/core/adapters/abcuploader.py:88]: If nothing is left in a method other than calling super, does it make sense to have it ?

File: framework_tvb/tvb/core/adapters/constants.py [hunk 2]
```diff
@@ -73,3 +73,8 @@
 
 ALL_TYPES = (TYPE_STR, TYPE_FLOAT, TYPE_INT, TYPE_UPLOAD, TYPE_BOOL, TYPE_DICT,
              TYPE_ARRAY, TYPE_SELECT, TYPE_MULTIPLE, TYPE_DYNAMIC, TYPE_LIST)
+
```
Comment (liadomide) [inline framework_tvb/tvb/core/adapters/constants.py:76]: A rebase with master would be nice to have, as I recently cleaned `constants.py`

File: framework_tvb/tvb/core/adapters/constants.py [hunk 3]
```diff
@@ -73,3 +73,8 @@
...
+ENCRYPTED_PASSWORD_NAME = 'encrypted_password.pem'
+ENCRYPTED_DATA_SUFFIX = '_encrypted'
+DECRYPTED_DATA_SUFFIX = '_decrypted'
```
Comment (liadomide) [inline framework_tvb/tvb/core/adapters/constants.py:79]: I know it was my suggestion to have the constants here, but now they only seem used inside `abcuploader.py`. Could they be better there ?

File: framework_tvb/tvb/core/adapters/abcuploader.py (hunk 1 above)
Comment (VinczeRobert) [reply framework_tvb/tvb/core/adapters/abcuploader.py:88]: No.

File: framework_tvb/tvb/core/adapters/constants.py (hunk 3 above)
Comment (VinczeRobert) [reply framework_tvb/tvb/core/adapters/constants.py]: Done.

**Pull request merged**